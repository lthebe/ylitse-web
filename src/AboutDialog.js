import React from 'react';
import PropTypes from 'prop-types';
import Button from 'material-ui/Button';
import Dialog, {
    DialogTitle,
    DialogContent,
    DialogContentText,
    DialogActions,
} from 'material-ui/Dialog';

// eslint-disable-next-line object-curly-newline
const AboutDialog = ({ uiVersion, apiVersion, onOkClick, ...props }) => (
    <Dialog {...props}>
        <DialogTitle>About</DialogTitle>
        <DialogContent>
            <DialogContentText>
                UI version: {uiVersion}
            </DialogContentText>
            <DialogContentText>
                API version: {apiVersion}
            </DialogContentText>
        </DialogContent>
        <DialogActions>
            <Button color="primary" onClick={onOkClick}>
                OK
            </Button>
        </DialogActions>
    </Dialog>
);

AboutDialog.propTypes = {
    uiVersion: PropTypes.string.isRequired,
    apiVersion: PropTypes.string.isRequired,
    onOkClick: PropTypes.func.isRequired,
};

export default AboutDialog;
