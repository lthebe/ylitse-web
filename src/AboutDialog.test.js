import React from 'react';
import { shallow } from 'enzyme';

import AboutDialog from './AboutDialog';

describe('AboutDialog', () => {
    test('renders correctly', () => {
        const props = {
            uiVersion: 'N/A', apiVersion: 'N/A', onOkClick: () => {},
        };
        const wrapper = shallow(<AboutDialog {...props} />);

        expect(wrapper).toMatchSnapshot();
    });
});
