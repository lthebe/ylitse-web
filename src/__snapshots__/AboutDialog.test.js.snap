// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutDialog renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AboutDialog
    apiVersion="N/A"
    onOkClick={[Function]}
    uiVersion="N/A"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <WithStyles(DialogTitle)>
          About
        </WithStyles(DialogTitle)>,
        <WithStyles(DialogContent)>
          <WithStyles(DialogContentText)>
            UI version: 
            N/A
          </WithStyles(DialogContentText)>
          <WithStyles(DialogContentText)>
            API version: 
            N/A
          </WithStyles(DialogContentText)>
        </WithStyles(DialogContent)>,
        <WithStyles(DialogActions)>
          <WithStyles(Button)
            color="secondary"
            onClick={[Function]}
          >
            OK
          </WithStyles(Button)>
        </WithStyles(DialogActions)>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "About",
        },
        "ref": null,
        "rendered": "About",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <WithStyles(DialogContentText)>
              UI version: 
              N/A
            </WithStyles(DialogContentText)>,
            <WithStyles(DialogContentText)>
              API version: 
              N/A
            </WithStyles(DialogContentText)>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                "UI version: ",
                "N/A",
              ],
            },
            "ref": null,
            "rendered": Array [
              "UI version: ",
              "N/A",
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                "API version: ",
                "N/A",
              ],
            },
            "ref": null,
            "rendered": Array [
              "API version: ",
              "N/A",
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <WithStyles(Button)
            color="secondary"
            onClick={[Function]}
          >
            OK
          </WithStyles(Button)>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "OK",
            "color": "secondary",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "OK",
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <WithStyles(DialogTitle)>
            About
          </WithStyles(DialogTitle)>,
          <WithStyles(DialogContent)>
            <WithStyles(DialogContentText)>
              UI version: 
              N/A
            </WithStyles(DialogContentText)>
            <WithStyles(DialogContentText)>
              API version: 
              N/A
            </WithStyles(DialogContentText)>
          </WithStyles(DialogContent)>,
          <WithStyles(DialogActions)>
            <WithStyles(Button)
              color="secondary"
              onClick={[Function]}
            >
              OK
            </WithStyles(Button)>
          </WithStyles(DialogActions)>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "About",
          },
          "ref": null,
          "rendered": "About",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <WithStyles(DialogContentText)>
                UI version: 
                N/A
              </WithStyles(DialogContentText)>,
              <WithStyles(DialogContentText)>
                API version: 
                N/A
              </WithStyles(DialogContentText)>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  "UI version: ",
                  "N/A",
                ],
              },
              "ref": null,
              "rendered": Array [
                "UI version: ",
                "N/A",
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  "API version: ",
                  "N/A",
                ],
              },
              "ref": null,
              "rendered": Array [
                "API version: ",
                "N/A",
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <WithStyles(Button)
              color="secondary"
              onClick={[Function]}
            >
              OK
            </WithStyles(Button)>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "OK",
              "color": "secondary",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "OK",
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
